File: WolfDen/wwwroot/ts/Behaviours/MoveBehaviour/NormalMoveBehaviourTemplate.ts
/// <reference path="../../Unit.ts" />

namespace TurnBasedBoardGame {
    export class NormalMoveBehaviourTemplate implements IBehaviourTemplate {
        
        Class: string;
        Speed: number = 0;
        MovementType: MovementType = MovementType.Foot;

        AddBehaviour(unit: Unit): void {
            let self = this;
            let movebehaviour = new NormalMoveBehaviour();
            movebehaviour.Unit = unit;
            movebehaviour.Speed = self.Speed;
            movebehaviour.Game = unit.Game;
            movebehaviour.MovementType = self.MovementType;
            movebehaviour.Id = unit.Id + self.Class;
            movebehaviour.Initialize();            
            unit.Behaviours[movebehaviour.Id] = movebehaviour;
        }
    }
}

File: WolfDen/Game/Behaviours/MoveBehaviour/NormalMoveBehaviourTemplate.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using WolfDen.Game.Behaviours.MoveBehaviour;

namespace WolfDen.Game.Templates
{
    public class NormalMoveBehaviourTemplate : IBehaviourTemplate
    {
        public string Class { get { return GetType().Name; } }
        public int Speed { get; set; } = 3;
        public MovementType MovementType { get; set; } = MovementType.Foot;

        public void AddBehaviour(Unit unit)
        {
            var moveBehaviour = new NormalMoveBehaviour(unit, unit.Game, Speed, MovementType);
            moveBehaviour.Id = unit.Id + Class;
            unit.Behaviours.Add(moveBehaviour.Id, moveBehaviour);
        }
    }
}